File: app/src/main/java/com/wj/myapplication/MainActivity.java
package com.wj.myapplication;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.xuexiang.xui.widget.guidview.FocusShape;
import com.xuexiang.xui.widget.guidview.GuideCaseView;
import com.xuexiang.xui.widget.guidview.OnViewInflateListener;


public class MainActivity extends BaseActivity {

    private Button button;
    private GuideCaseView guideCaseView;
    @Override
    protected int setLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    protected void initView() {
        button = findViewById(R.id.button);
    }

    @Override
    protected void initData() {
        guideCaseView = new GuideCaseView.Builder(this)
                .focusOn(button)
                .focusShape(FocusShape.ROUNDED_RECTANGLE)
                .roundRectRadius(20)
                .disableFocusAnimation()
                .customView(R.layout.layout_custom_guide_case_view, new OnViewInflateListener() {
                    @Override
                    public void onViewInflated(View view12) {
                        view12.findViewById(R.id.btn_action_close).setOnClickListener(new View.OnClickListener() {
                            @Override
                            public void onClick(View view1) {
                                guideCaseView.hide();
                            }
                        });
                    }
                })
                .closeOnTouch(false)
                .build();
        guideCaseView.show();
    }

    @Override
    protected void setEvent() {
//        button.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//                doSomeThing();
//            }
//        });
    }

    private void doSomeThing() {

    }
}

File: app/src/main/java/com/wj/myapplication/MainActivity.kt
package com.wj.myapplication

import android.Manifest
import android.annotation.SuppressLint
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.Toast

import com.tbruyelle.rxpermissions2.Permission
import com.tbruyelle.rxpermissions2.RxPermissions

import java.util.regex.Matcher
import java.util.regex.Pattern

import io.reactivex.functions.Consumer

class MainActivity : BaseActivity() {

    var button: Button? = null
    var but1: Button? = null

    override fun setLayoutId(): Int {
        return R.layout.activity_main
    }

    override fun initView() {
        button = findViewById(R.id.button)
    }

    override fun initData() {}

    @SuppressLint("ClickableViewAccessibility")
    override fun setEvent() {
        button!!.setOnClickListener {
            doSomeThing() }

        button!!.setOnTouchListener { v, event ->
            v.layoutParams
            event.action
            event.rawX
            false
        }
        button!!.setOnTouchListener{q,w ->
            false
        }

    }

    @SuppressLint("CheckResult")
    fun doSomeThing() {
        val rxPermissions = RxPermissions(this)
        rxPermissions.requestEach(Manifest.permission.READ_SMS, Manifest.permission.RECEIVE_SMS)
                .subscribe { permission ->
                    if (permission.name.equals(Manifest.permission.ACCESS_FINE_LOCATION, ignoreCase = true)) {
                        if (permission.granted) {//同意后调用
                        } else if (permission.shouldShowRequestPermissionRationale) {//禁止，但没有选择“以后不再询问”，以后申请权限，会继续弹出提示
                        } else {//禁止，但选择“以后不再询问”，以后申请权限，不会继续弹出提示
                        }
                    }
                }
        Toast.makeText(this, "hello world!", Toast.LENGTH_SHORT).show()
    }

}
